src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,21 @@
+// child class of Popup
+
+import Popup from "./Popup.js";
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleFormSubmit, confirmationButtonSelector) {
+    super(popupSelector);
+    this._handleFormSubmit = handleFormSubmit;
```

Comment:
`PopupWithConfirmation` is a general class. It can be used for any confirmation and there should be only one instance of this class. This class should have a setter for `_handleFormSubmit`, so you can set the action when opening the confirm for the specific card

```js
constructor(popupSelector) {
  super(popupSelector);
  this._form = this._modal.querySelector(".modal__form");
}

setAction(action) { 
  this._handleFormSubmit = action; 
} 

 this._confirmationButton.addEventListener("click", this._handleFormSubmit);
```